aws-distro-opentelemetry-node-autoinstrumentation/src/patches/aws/services/bedrock.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Attributes, DiagLogger, Span, SpanKind, Tracer } from '@opentelemetry/api';
+import {
+  AwsSdkInstrumentationConfig,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWS_ATTRIBUTE_KEYS } from '../../../aws-attribute-keys';
+import { RequestMetadata, ServiceExtension } from '../../../third-party/otel/aws/services/ServiceExtension';
+import { AwsSpanProcessingUtil } from '../../../aws-span-processing-util';
+
+const AGENT_ID: string = 'agentId';
+const KNOWLEDGE_BASE_ID: string = 'knowledgeBaseId';
+const DATA_SOURCE_ID: string = 'dataSourceId';
+const GUARDRAIL_ID: string = 'guardrailId';
+const MODEL_ID: string = 'modelId';
+const AWS_BEDROCK_SYSTEM: string = 'aws_bedrock';
+
+const AGENT_OPERATIONS = [
+  'CreateAgentActionGroup',
+  'CreateAgentAlias',
+  'DeleteAgentActionGroup',
+  'DeleteAgentAlias',
+  'DeleteAgent',
+  'DeleteAgentVersion',
+  'GetAgentActionGroup',
+  'GetAgentAlias',
+  'GetAgent',
+  'GetAgentVersion',
+  'ListAgentActionGroups',
+  'ListAgentAliases',
+  'ListAgentKnowledgeBases',
+  'ListAgentVersions',
+  'PrepareAgent',
+  'UpdateAgentActionGroup',
+  'UpdateAgentAlias',
+  'UpdateAgent',
+];
+
+const KNOWLEDGE_BASE_OPERATIONS = [
+  'AssociateAgentKnowledgeBase',
+  'CreateDataSource',
+  'DeleteKnowledgeBase',
+  'DisassociateAgentKnowledgeBase',
+  'GetAgentKnowledgeBase',
+  'GetKnowledgeBase',
+  'ListDataSources',
+  'UpdateAgentKnowledgeBase',
+];
+
+const DATA_SOURCE_OPERATIONS = ['DeleteDataSource', 'GetDataSource', 'UpdateDataSource'];
+
+const operationToBedrockAgentServiceMapping: { [key: string]: { [key: string]: string } } = {};
```

Comment:
Can we change var name to something more explicit, like `operationToBedrockAgentAttributesMap`.

Also, since this is a mapping from operations to a map of attributes, we can also change type from `{ [key: string]: { [key: string]: string } }` to the `AttributeMap` type from `metric-attribute-generator.ts`
```
const operationToBedrockAgentAttributesMap: AttributeMap = {};
```

Reply:
Renamed but did not change the type.

Changing it to AttributeMap also doesn't really make sense because it's not a attribute key to attribute value map, it's a request/response attribute key to span attribute key map. Different use case

We would also need to add a bunch of logic to convert the attributes to strings whenever we want to use them since in this case they're always strings and we use them as such.